vault/core.go

```diff
@@ -1967,6 +1968,7 @@ func (s standardUnsealStrategy) unseal(ctx context.Context, logger log.Logger, c
 	if err := c.setupMounts(ctx); err != nil {
 		return err
 	}
+	enterpriseSetupAPILock(c)
```

Comment:
Should this be able to return an error?

Reply:
i was trying to keep the API lock init error free. there isn't much in the setup to error, unless we want to be really defensive and error if the logger or ent namespace manager or something don't exist (which are already set up before this is - on ent)